front/pages/api/w/[wId]/assistant/conversations/[cId]/messages/[mId]/upload_raw_content_fragment.ts

```diff
@@ -0,0 +1,186 @@
+        type: "workspace_not_found",
+        message: "The workspace you're trying to modify was not found.",
+      },
+    });
+  }
+
+  const user = auth.user();
+  if (!user) {
+    return apiError(req, res, {
+      status_code: 404,
+      api_error: {
+        type: "workspace_user_not_found",
+        message: "Could not find the user of the current session.",
+      },
+    });
+  }
+
+  if (!auth.isUser()) {
+    return apiError(req, res, {
+      status_code: 403,
+      api_error: {
+        type: "workspace_auth_error",
+        message:
+          "Only users of the current workspace can update chat sessions.",
+      },
+    });
+  }
+  if (!(typeof req.query.cId === "string")) {
+    return apiError(req, res, {
+      status_code: 400,
+      api_error: {
+        type: "invalid_request_error",
+        message: "Invalid query parameters, `cId` (string) is required.",
+      },
+    });
+  }
+
+  const conversationId = req.query.cId;
+  const conversation = await getConversation(auth, conversationId);
+  if (!conversation) {
+    return apiError(req, res, {
+      status_code: 404,
+      api_error: {
+        type: "conversation_not_found",
+        message: "Conversation not found.",
+      },
+    });
+  }
+
+  if (!(typeof req.query.mId === "string")) {
+    return apiError(req, res, {
+      status_code: 400,
+      api_error: {
+        type: "invalid_request_error",
+        message: "Invalid query parameters, `mId` (string) is required.",
+      },
+    });
+  }
+  const messageId = req.query.mId;
+
+  switch (req.method) {
+    case "POST":
+      const message = conversation.content
+        .flat()
+        .find((m) => m.sId === messageId);
+      if (!message || !isContentFragmentType(message)) {
+        return apiError(req, res, {
+          status_code: 400,
+          api_error: {
+            type: "invalid_request_error",
+            message:
+              "Uploading raw content fragment is only supported for 'content fragment' messages.",
+          },
+        });
+      }
+      try {
+        const form = new IncomingForm();
+        const [_fields, files] = await form.parse(req);
+        void _fields;
+
+        const maybeFiles = files.file;
+
+        if (!maybeFiles) {
+          return apiError(req, res, {
+            status_code: 400,
+            api_error: {
+              type: "invalid_request_error",
+              message: "No file uploaded",
+            },
+          });
+        }
+
+        const file = maybeFiles[0];
+
+        const storage = new Storage({
+          keyFilename: SERVICE_ACCOUNT,
+        });
+
+        const bucket = storage.bucket(DUST_PRIVATE_UPLOADS_BUCKET);
+        const filePath = `content_fragments/w/${owner.sId}/assistant/conversations/${conversation.sId}/content_fragment/${message.sId}/raw`;
+        const gcsFile = bucket.file(filePath);
+        const fileStream = fs.createReadStream(file.filepath);
+
+        await new Promise((resolve, reject) =>
```

Comment:
No need for this, there is a Stream Promise API in our Node version: https://nodejs.org/api/stream.html#streampipelinesource-transforms-destination-options

Here is an example:

https://github.com/dust-tt/dust/blob/1973f6be8a4134ea182b2b8c9abbc87a586041c1/connectors/src/connectors/github/lib/github_api.ts#L693